format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatProvider.java

```diff
@@ -0,0 +1,192 @@
+import io.cdap.cdap.api.plugin.PluginClass;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.validation.FormatContext;
+import io.cdap.cdap.etl.api.validation.InputFile;
+import io.cdap.cdap.etl.api.validation.InputFiles;
+import io.cdap.cdap.etl.api.validation.ValidatingInputFormat;
+import io.cdap.plugin.format.input.PathTrackingConfig;
+import io.cdap.plugin.format.input.PathTrackingInputFormatProvider;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.DataFormatter;
+import org.apache.poi.ss.usermodel.FormulaEvaluator;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.apache.poi.ss.util.CellReference;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+
+/**
+ * Reads XLS(X) into StructuredRecords.
+ */
+@Plugin(type = ValidatingInputFormat.PLUGIN_TYPE)
+@Name(XlsInputFormatProvider.NAME)
+@Description(XlsInputFormatProvider.DESC)
+public class XlsInputFormatProvider extends PathTrackingInputFormatProvider<XlsInputFormatConfig> {
+  static final String NAME = "xls";
+  static final String DESC = "Plugin for reading files in xls(x) format.";
+  public static final PluginClass PLUGIN_CLASS = PluginClass.builder()
+          .setType(ValidatingInputFormat.PLUGIN_TYPE)
+          .setName(NAME)
+          .setDescription(DESC)
+          .setClassName(XlsInputFormatProvider.class.getName())
+          .setConfigFieldName("conf")
+          .setProperties(XlsInputFormatConfig.XLS_FIELDS)
+          .build();
+  private final XlsInputFormatConfig conf;
+
+  public XlsInputFormatProvider(XlsInputFormatConfig conf) {
+    super(conf);
+    this.conf = conf;
+  }
+
+  @Override
+  public String getInputFormatClassName() {
+    return XlsInputFormat.class.getName();
+  }
+
+  @Override
+  public void validate(FormatContext context) {
+    Schema schema = super.getSchema(context);
+    // When the sheet is specified by number, the sheet value must be a number
+    if (!conf.containsMacro(XlsInputFormatConfig.NAME_SHEET_VALUE)
+            && conf.getSheet().equals(XlsInputFormatConfig.SHEET_NUMBER)
+                    && (conf.getSheetValue() == null || !conf.getSheetValue().matches("[0-9]+"))) {
+      context.getFailureCollector().addFailure("Sheet number must be a number.", null)
+              .withConfigProperty(XlsInputFormatConfig.NAME_SHEET_VALUE);
+    }
+    FailureCollector collector = context.getFailureCollector();
+    if (!conf.containsMacro(PathTrackingConfig.NAME_SCHEMA) && schema == null && context.getInputSchema() == null) {
+      collector.addFailure("XLS format cannot be used without specifying a schema.", "Schema must be specified.")
```

Comment:
be consistent with line#85 or make use of `collector` both the places. 
Wondering when the `collector.getOrThrowException()` is called.

Reply:
Using `collector` for both.
`collector.getOrThrowException()` is called internally in `configureStage` and `getSchema` of parent class.